src/foam/mlang/expr/Ref.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.mlang.expr',
+  name: 'Ref',
+  extends: 'foam.mlang.AbstractExpr',
+  implements: [ 'foam.core.Serializable' ],
+
+  documentation: 'An Unary Predicate which returns reference property object',
+
+  javaImports: [
+    'foam.core.FObject',
+    'foam.core.PropertyInfo',
+    'foam.nanos.logger.Logger',
+    'foam.nanos.logger.StdoutLogger',
+    'foam.util.StringUtil'
+  ],
+
+  properties: [
+    {
+      class: 'foam.mlang.ExprProperty',
+      name: 'arg1'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'f',
+      code: async function(o) {
```

Comment:
JS predicates can't be async. Better to throw an exception.

Reply:
fixed

Reply:
Don't make "async".

Reply:
fixed